urlaeeaf948fe_smcpeak_ded_tgz-pJ8-ded_model_SerializationTestsJ8/src/util/awt/AWTJSONUtil.java

```diff
@@ -0,0 +1,89 @@
+// AWTJSONUtil.java
+// See toplevel license.txt for copyright and license terms.
+
+package util.awt;
+
+import java.awt.Dimension;
+import java.awt.Point;
+import java.util.EnumSet;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/** Utilities related to serializing AWT objects as JSON. */
+public class AWTJSONUtil {
+    /** Serialize 'p' as JSON. */
+    public static JSONObject pointToJSON(Point p)
+    {
+        JSONObject o = new JSONObject();
+        try {
+            o.put("x", p.x);
+            o.put("y", p.y);
+        }
+        catch (JSONException e) { assert(false); }
+        return o;
+    }
+
+    /** Deserialize a Point from 'o'. */
+    public static Point pointFromJSON(JSONObject o) throws JSONException
+    {
+        Point p = new Point();
+        p.x = (int)o.getLong("x");
+        p.y = (int)o.getLong("y");
+        return p;
+    }
+
+    /** Serialize 'd' as JSON. */
+    public static JSONObject dimensionToJSON(Dimension d)
+    {
+        JSONObject o = new JSONObject();
+        try {
+            o.put("w", d.width);
+            o.put("h", d.height);
+        }
+        catch (JSONException e) { assert(false); }
+        return o;
+    }
+
+    /** Deserialize a Dimension from 'o'. */
+    public static Dimension dimensionFromJSON(JSONObject o) throws JSONException
+    {
+        Dimension d = new Dimension();
+        d.width = (int)o.getLong("w");
+        d.height = (int)o.getLong("h");
+        return d;
+    }
+
+    /** Render 'set' as a JSON array. */
+    public static <T extends Enum<T>> JSONArray enumSetToJSON(EnumSet<T> set)
+    {
+        JSONArray ret = new JSONArray();
+        for (T t : set) {
+            ret.put(t.name());
+        }
+        return ret;
+    }
+
+    /** Deserialize an EnumSet from a JSON array. */
+    public static <T extends Enum<T>>
+    EnumSet<T> enumSetFromJSON(Class<T> enumType, JSONArray array) throws JSONException
+    {
+        EnumSet<T> ret = EnumSet.noneOf(enumType);
+        for (int i=0; i < array.length(); i++) {
+            String element = array.getString(i);
+            try {
+                T t = Enum.valueOf(enumType, element);
+                ret.add(t);
+            }
+            catch (IllegalArgumentException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)